src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -219,6 +238,19 @@ public boolean isSatisfiedBy(ByteBuffer columnValue)
 
         Value value = new Value(columnValue, validator);
 
+        if (operation == Op.BOUNDED_ANN)
```

Comment:
can we leverage GeoDistanceExpression code here?  (If that requires weird contortions, fine to leave it as is)

Reply:
This raises a larger question about why we have these two classes which have seemingly similar logic and names: `org.apache.cassandra.index.sai.plan.Expression` and `org.apache.cassandra.db.filter.RowFilter.Expression`. Both classes have duplicated `isSatisfiedBy` logic. I am going to defer on de-duping the logic for now, since it seems like a somewhat intentional design choice.

One possible option is to create yet another class that contains the `GeoDistanceExpression` logic for satisfaction.

Reply:
I asked MikeA once and he said, effectively, "we should merge these but we haven't had time yet."

Fine to defer for now.